soc/xtensa/espressif_esp32/esp32/default.ld

```diff
@@ -453,6 +453,8 @@ SECTIONS
     *libsubsys__net__ip.a:(.noinit .noinit.*)
     *libsubsys__net.a:(.noinit .noinit.*)
 #endif
+    _spiram_heap_start = ABSOLUTE(.);
+    . = . + CONFIG_SPIRAM_HEAP_SIZE;
 
     *(.ext_ram.bss*)
     _ext_ram_data_end = ABSOLUTE(.) + CONFIG_ESP_SPIRAM_SIZE;
```

Comment:
@EricNRS, I've been thinking about this and I suggest adding 2 changes in this PR (for all SoCs):

1) `_ext_ram_data_end = ABSOLUTE(.) + CONFIG_ESP_SPIRAM_SIZE;` should be changed to
`_ext_ram_data_end = ABSOLUTE(.);` only. As is, total external ram usage will wrongly indicate more that available.  
2) Would you also add some `ASSERT()` in there so we check whether SPIRAM region was overflowed:
```c
ASSERT(((_ext_ram_data_end - _ext_ram_data_start) <= CONFIG_ESP_SPIRAM_SIZE),
          "External SPIRAM overflowed.")
```

Reply:
You are correct, `_ext_ram_data_end = ABSOLUTE(.) + CONFIG_ESP_SPIRAM_SIZE;` is wrong and should be changed to `_ext_ram_data_end = ABSOLUTE(.) ` since it is double-allocating the CONFIG_ESP_SPIRAM_SIZE.  The assert is a great idea.

I'll fix that and upload the changes tomorrow.

Reply:
Done, please take another look when you have a chance.

I have tested on ESP32-S3 and ESP32.  Can you test on an ESP-S2?